src/components/Swap/SwapButton/index.tsx

```diff
@@ -83,21 +83,22 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
         chainId,
       })
       setDisplayTxHash(transaction.hash)
+      // Only reset pending after any queued animations to avoid layout thrashing, because a
+      // successful wrap will open the status dialog and immediately cover the button.
+      const postWrap = () => {
+        setIsPending(false)
+        document.removeEventListener('animationend', postWrap)
+      }
+      if (isAnimating(document)) {
+        document.addEventListener('animationend', postWrap)
+      } else {
+        postWrap()
```

Comment:
This can just be `setIsPending(false)`, because it doesn't need to remove an event listener.